app/src/main/java/com/example/randomusergenerator/user/UserScreen.kt

```diff
@@ -52,7 +48,9 @@ fun UserListScreen(userViewState: UserViewState) {
             userViewState.users?.let { userList ->
                 LazyColumn(modifier = Modifier.fillMaxSize()) {
                     items(userList) { user ->
-                        UserCard(user)
+                        UserCard(user, onClick = {
+                            navController.navigate("user_details_screen/${it.id}")
```

Comment:
Remember state hoisting? You could use the High Order Function instead of passing down the navcontroller.